echo server qemu_x86 e1000 crash when coverage is enabled

echo server qemu_x86 e1000 crashes if coverage is enabled. To reproduce:

Compile and run echo server
```
cd samples/net/sockets/echo_server/build
cmake -DBOARD=qemu_x86 -DOVERLAY_CONFIG=overlay-e1000.conf -DCONFIG_COVERAGE=y ..
make
make run

```
After couple of seconds it crashes:

```
[00:00:03.180,000] <inf> net_echo_server_sample: Waiting for TCP connection on port 4242 (IPv6)...
[00:00:03.180,000] <err> os: EAX: 0x00000001, EBX: 0xaaaaaaaa, ECX: 0xaaaaaaaa, EDX: 0x00228b80
[00:00:03.180,000] <err> os: ESI: 0xaaaaaaaa, EDI: 0x0010937c, EBP: 0xaaaaaaaa, ESP: 0x00236ecc
[00:00:03.180,000] <err> os: EFLAGS: 0x00041b03 CS: 0x0008 CR3: 0x002272e0
[00:00:03.180,000] <err> os: call trace:
[00:00:03.180,000] <err> os: EIP: 0x00236ee8
[00:00:03.180,000] <err> os: unaligned frame ptr
[00:00:03.180,000] <err> os: >>> ZEPHYR FATAL ERROR 1: Unhandled interrupt on CPU 0
[00:00:03.180,000] <err> os: Current thread: 0x00226120 (unknown)
[00:00:03.190,000] <err> os: Halting system
make[3]: *** [zephyr/CMakeFiles/run.dir/build.make:60: zephyr/CMakeFiles/run] Error 1
make[2]: *** [CMakeFiles/Makefile2:622: zephyr/CMakeFiles/run.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:629: zephyr/CMakeFiles/run.dir/rule] Error 2
make: *** [Makefile:274: run] Error 2
```

It still crashes with memory increased:

```
CONFIG_MAIN_STACK_SIZE=8192
CONFIG_NET_PKT_RX_COUNT=160
CONFIG_NET_PKT_TX_COUNT=160
CONFIG_NET_BUF_RX_COUNT=640
CONFIG_NET_BUF_TX_COUNT=640
```
echo server qemu_x86 does not crash if -DCONFIG_COVERAGE=y is not passed to cmake
Also, echo server does not crash with coverage enabled and board native_posix

--- a/samples/net/sockets/echo_server/src/common.h
+++ b/samples/net/sockets/echo_server/src/common.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017 Intel Corporation.
+ * Copyright (c) 2017-2019 Intel Corporation.
  * Copyright (c) 2018 Nordic Semiconductor ASA.
  *
  * SPDX-License-Identifier: Apache-2.0
@@ -7,7 +7,8 @@
 
 
 #define MY_PORT 4242
-#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS) || defined(CONFIG_NET_TCP2)
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS) || defined(CONFIG_NET_TCP2) || \
+	defined(CONFIG_COVERAGE)
 #define STACK_SIZE 4096
 #else
 #define STACK_SIZE 1024

